File: Easy/istanza/meta_istanzaeq_segview/meta_istanzaeq_segview.cs

/*
Easy
Copyright (C) 2021 Università degli Studi di Catania (www.unict.it)
This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.
This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.
You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/


using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_istanzaeq_segview
{
    public class Meta_istanzaeq_segview : Meta_easydata 
	{
        public Meta_istanzaeq_segview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "istanzaeq_segview") {
				Name = "Istanza di equipollenza ";
			EditTypes.Add("eq_seg");
			ListingTypes.Add("eq_seg");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idistanza","idistanzakind","idreg_studenti"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "eq_seg": {
						DescribeAColumn(T, "registrystudenti_title", "Studente", nPos++);
						DescribeAColumn(T, "aa", "Anno accademico", nPos++);
						DescribeAColumn(T, "istanza_data", "Data", nPos++);
						if (T.Columns.Contains("istanza_data")) T.Columns["istanza_data"].ExtendedProperties["format"] = "g";
						DescribeAColumn(T, "corsostudio_title", "Corso di studi", nPos++);
						DescribeAColumn(T, "didprog_title", "Denominazione Didattica programmata", nPos++);
						DescribeAColumn(T, "didprog_aa", "Anno accademico Didattica programmata", nPos++);
						DescribeAColumn(T, "didprog_idsede", "Sede Didattica programmata", nPos++);
						DescribeAColumn(T, "iscrizione_anno", "Anno di corso Iscrizione", nPos++);
						DescribeAColumn(T, "iscrizione_iddidprog", "Didattica programmata Iscrizione", nPos++);
						DescribeAColumn(T, "iscrizione_aa", "Anno accademico Iscrizione", nPos++);
						DescribeAColumn(T, "statuskind_title", "Status", nPos++);
						DescribeAColumn(T, "istanzaparent_idistanzakind", "Tipologia Istanza collegata", nPos++);
						DescribeAColumn(T, "istanzaparent_idreg_studenti", "Studente Istanza collegata", nPos++);
						DescribeAColumn(T, "istanzaparent_aa", "Anno accademico Istanza collegata", nPos++);
						DescribeAColumn(T, "istanzaparent_data", "Data Istanza collegata", nPos++);
						if (T.Columns.Contains("istanzaparent_data")) T.Columns["istanzaparent_data"].ExtendedProperties["format"] = "g";
						DescribeAColumn(T, "istanzaparent_iddidprog", "Didattica programmata Istanza collegata", nPos++);
						DescribeAColumn(T, "istanzaparent_idstatuskind", "Status Istanza collegata", nPos++);
						DescribeAColumn(T, "istanzaparent_idiscrizione", "Iscrizione Istanza collegata", nPos++);
						DescribeAColumn(T, "dichiartitolo_seg_iddichiarkind", "Dichiarazione del titolo di studio", nPos++);
						DescribeAColumn(T, "dichiartitolo_seg_idreg", "Dichiarazione del titolo di studio", nPos++);
						DescribeAColumn(T, "dichiartitolo_seg_aa", "Dichiarazione del titolo di studio", nPos++);
						DescribeAColumn(T, "dichiartitolo_seg_date", "Dichiarazione del titolo di studio", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		//$GetSortings$

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/istanza/meta_istanzaeq_segview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_istanzaeq_segview() {
        MetaData.apply(this, ["istanzaeq_segview"]);
        this.name = 'meta_istanzaeq_segview';
    }

    meta_istanzaeq_segview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_istanzaeq_segview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'eq_seg':
						this.describeAColumn(table, 'registrystudenti_title', 'Studente', null, 1300, 101);
						this.describeAColumn(table, 'aa', 'Anno accademico', null, 2000, 9);
						this.describeAColumn(table, 'istanza_data', 'Data', 'g', 3000, null);
						this.describeAColumn(table, 'didprog_title', 'Denominazione Didattica equipollente', null, 6100, 1024);
						this.describeAColumn(table, 'didprog_aa', 'Anno accademico Didattica equipollente', null, 6200, 9);
						this.describeAColumn(table, 'sede_title', 'Denominazione Sede Didattica equipollente', null, 6320, 1024);
						this.describeAColumn(table, 'statuskind_title', 'Status', null, 10200, 50);
						this.describeAColumn(table, 'istanza_protanno', 'Anno di protocollo', null, 12000, null);
						this.describeAColumn(table, 'istanza_protnumero', 'Numero di protocollo', null, 13000, null);
						this.describeAColumn(table, 'dichiartitolo_seg_aa', 'Dichiarazione del titolo di studio', null, 109100, 9);
						this.describeAColumn(table, 'dichiartitolo_seg_date', 'Dichiarazione del titolo di studio', null, 109300, null);
						this.describeAColumn(table, 'dichiartitolo_seg_iddichiarkind', 'Dichiarazione del titolo di studio', null, 109500, null);
						this.describeAColumn(table, 'dichiartitolo_seg_idreg', 'Dichiarazione del titolo di studio', null, 109600, null);
//$objCalcFieldConfig_eq_seg$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idistanza", "idistanzakind", "idreg_studenti"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			//$getSorting$

        });

    window.appMeta.addMeta('istanzaeq_segview', new meta_istanzaeq_segview('istanzaeq_segview'));

	}());
